src/main/java/com/mongodb/spark/sql/connector/read/MongoMicroBatchPartitionReader.java

```diff
@@ -0,0 +1,191 @@
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.model.Aggregates;
+import com.mongodb.client.model.Filters;
+
+import com.mongodb.spark.sql.connector.assertions.Assertions;
+import com.mongodb.spark.sql.connector.config.ReadConfig;
+import com.mongodb.spark.sql.connector.exceptions.MongoSparkException;
+import com.mongodb.spark.sql.connector.schema.BsonDocumentToRowConverter;
+
+/**
+ * A partition reader returned by {@link
+ * MongoPartitionReaderFactory#createReader(org.apache.spark.sql.connector.read.InputPartition)}.
+ * It's responsible for outputting data for a RDD partition.
+ */
+public class MongoMicroBatchPartitionReader implements PartitionReader<InternalRow> {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(MongoMicroBatchPartitionReader.class);
+  private static final String FULL_DOCUMENT = "fullDocument";
+  private final MongoMicroBatchInputPartition partition;
+  private final BsonDocumentToRowConverter bsonDocumentToRowConverter;
+  private final ReadConfig readConfig;
+  private final MongoClient mongoClient;
+  private volatile boolean closed = false;
+  private MongoChangeStreamCursor<BsonDocument> changeStreamCursor;
+  private InternalRow currentRow;
+
+  /**
+   * Construct a new instance
+   *
+   * @param partition the partition
+   * @param bsonDocumentToRowConverter the converter from {@link BsonDocument} to {@link
+   *     InternalRow}
+   * @param readConfig the read configuration for reading from the partition
+   */
+  public MongoMicroBatchPartitionReader(
+      final MongoMicroBatchInputPartition partition,
+      final BsonDocumentToRowConverter bsonDocumentToRowConverter,
+      final ReadConfig readConfig) {
+
+    this.partition = partition;
+    this.bsonDocumentToRowConverter = bsonDocumentToRowConverter;
+    this.readConfig = readConfig;
+    this.mongoClient = readConfig.getMongoClient();
+    LOGGER.info(
+        "Creating partition reader for: PartitionId: {} with Schema: {}",
+        partition.getPartitionId(),
+        bsonDocumentToRowConverter.getSchema());
+  }
+
+  /**
+   * Proceed to next record, returns false if there is no more records.
+   *
+   * <p>Enter a busy loop until one of the following scenarios exits the loop:
+   *
+   * <ul>
+   *   <li>There is a result returned from {@code cursor.tryNext()}.
+   *   <li>The cursor is closed. Happens when the collection is dropped.
+   *   <li>The postBatchResumeToken's timestamp from the change stream is after the
+   *       endOffsetTimestamp. Indicating that newer events are in the oplog but none within this
+   *       time period for this collection.
+   * </ul>
+   */
+  @Override
+  public boolean next() {
+    Assertions.ensureState(() -> !closed, () -> "Cannot call next() on a closed PartitionReader.");
+
+    MongoChangeStreamCursor<BsonDocument> cursor = getCursor();
+    BsonDocument cursorNext;
+
+    do {
+      cursorNext = cursor.tryNext();
```

Comment:
Do we need to wrap any `MongoException` thrown by `tryNext`, as we do in the `getCursor` method below?

Reply:
Ack - sound good.